README.md

```diff
@@ -90,15 +90,48 @@ Using the library involves data types and their mappings from Sophia to JavaScri
 | tuple               | `(true, false)`                                             | Array           | `[true, false]`                                                                |
 | map                 | `{[7] = false}`                                             | Map             | `new Map([[7, false]])`                                                        |
 | record              | `{x = 0, y = 0}`                                            | Object (POJO)   | `{x: 0, y: 0}`                                                                 |
-| variant             | `Some(404)`                                                 | Object (POJO)   | `{variant: 'Some', values: [404]}`                                             |
+| variant             | `Some(404)`, `None`                                         | Object (POJO)   | `{'Some': [404]}`, `{'None': []}`                                              |
 | bits                | `Bits.none`, `Bits.all`  `Bits.set(Bits.none, 0)`           | BigInt          | `0b0n`, `-1n`, `0b00000001n`                                                   |
 | hash                | `#001234d`                                                  | BigInt          | `BigInt("0x001234d")`                                                          |
 | signature           | `#001234d`                                                  | BigInt          | `BigInt("0x001234d")`                                                          |
 | address             | `ak_2gx9MEFxKvY9vMG5YnqnXWv1hCsX7rgnfvBLJS4aQurustR1rt`     | BigInt, String  | `BigInt("0xDE68BFE1B203E51F52351BA087F79B7828E6A140F0C314A670C7003B3FF57075")`, `ak_2gx9MEFxKvY9vMG5YnqnXWv1hCsX7rgnfvBLJS4aQurustR1rt` |
 
-- note the fixed structure of variant object with keys `variant` which denotes variant constructor and `values` as it's arguments.
+- note the fixed structure of variant object with a single key - the variant constructor (i.e. ``Some`) and array of variant arguments as it's value.
 - while Javascript Number and primitive `int` types can be used as well when `BigInt` type is expected it's not recommended because of it's `Number.MAX_SAFE_INTEGER` limitation.
 
+## Data constructors
+
+To easy `Variant` data construction this library provide the following constructors:
+
+- `Variant(constructor, arg1, arg2, ...argN)`
+- `Some(arg)`
+- `None()`
+
+Example:
+
+Given the following Sophia entrypoint:
+```sophia
+entrypoint test_optional(a:option(int)) = a
+```
+
+Below encoder call groups are equivalent:
```

Comment:
I'm not a fan of allowing to do the same thing in different ways. If Variant would be a class then we can define a corresponding argument as an instance of Variant class, this can help when adding TypeScript support. It is ok to have it as a plain object, but then a `Variant` helper function is not necessary.

Reply:
You asked for helpers. I won't mix public with internal interfaces of this lib, thus the new data constructors. I can live without them.

There is no plans to add any compilation step dependancies to this lib, like typescript or anything else unnecessary.

Reply:
TBH that constructor should be implemented upstream in general